src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect, useState } from 'react';
+import { SearchResultType } from '../components/SearchResults/SearchResultsTypes';
+import { useNavigate } from 'react-router-dom';
+import { APISearch, APICharacterID } from '../ulits/api';
+import SearchBar from '../components/SearchBar/SearchBar';
+import SearchResults from '../components/SearchResults/SearchResults';
+import { placeholderText, currentItemIDInitial } from './SearchPageVariables';
+import Pagination from '../components/Pagination/Pagination';
+import ItemsSelect from '../components/ItemsPerPage/ItemsSelect';
+import DetailsSection from '../components/DetailsSection/DetailsSection';
+
+export function SearchPage() {
+  const [query, setQuery] = useState('');
+  const [searchResults, setsearchResults] = useState<SearchResultType[]>([]);
+  const [isSearchLoading, setIsSearchLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [totalPages, setTotalPages] = useState(0);
+  const [limit, setLimit] = useState(10);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [openDetails, setOpenDetails] = useState(false);
+  const [currentItemID, setCurrentItemID] = useState('');
+  const [currentItem, setCurrentItem] =
+    useState<SearchResultType>(currentItemIDInitial);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const savedQuery = localStorage.getItem('searchQuery');
+    if (savedQuery) {
+      getSearchResults(savedQuery, limit, currentPage);
+    } else getSearchResults(query, limit, currentPage);
```

Comment:
You do the same on line 30 and 31
I guess, it will be more readable 
`const savedQuery = localStorage.getItem('searchQuery');`
`const queryString =  savedQuery  || query;` or similar `const queryString =  savedQuery ? savedQuery  : query` 
`getSearchResults(queryString , limit, currentPage);`